kong/db/declarative/init.lua

```diff
@@ -584,8 +581,8 @@ function declarative.load_into_cache(entities, meta, hash, shadow)
   -- but filtered for a given tag
   local tags_by_name = {}
 
-  kong.core_cache:purge(shadow)
-  kong.cache:purge(shadow)
+  local t = txn.begin(128)
```

Comment:
If we add the lock, It might make more sense to move it to the top of the load operation ("block the API, not the transaction mechanism"). This way we save memory not allocating and parsing massive JSON files just to block immediately afterwards.

Reply:
From the discussion [here](https://github.com/Kong/kong/pull/8224/files#r784737751), this doesn't seem to be a blocker to merge the PR, but a potential improvement. That's correct @kikito ?

Reply:
@Suika-Kong Already added some concurrency control for LMDB config loads, so I think this is less of a problem at this point.